--- eucalyptus-a13m/clc/modules/msgs/src/main/java/com/eucalyptus/records/Logs.java	2012-06-20 14:56:25.085224664 -0400
+++ eucalyptus-3.1.0/clc/modules/msgs/src/main/java/com/eucalyptus/records/Logs.java	2012-08-06 14:06:20.635101868 -0400
@@ -2,6 +2,7 @@
 
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
+import java.io.InputStream;
 import java.io.OutputStream;
 import java.io.PrintStream;
 import java.net.URL;
@@ -132,6 +133,10 @@
       arg1.getRootLogger( ).addAppender( console );
     }
     
+    @Override
+    public void doConfigure( final InputStream arg0, final LoggerRepository arg1 ) {
+      arg1.getRootLogger( ).addAppender( console );
+    }
   }
   
   private static final Logger nullLogger = new Logger( "/dev/null" ) {
