diff -ur eucalyptus-3.1.0.orig/clc/eucadmin/eucadmin/check.py eucalyptus-3.1.0/clc/eucadmin/eucadmin/check.py
--- eucalyptus-3.1.0.orig/clc/eucadmin/eucadmin/check.py	2012-06-19 14:41:48.000000000 -0400
+++ eucalyptus-3.1.0/clc/eucadmin/eucadmin/check.py	2012-08-16 16:32:39.180732968 -0400
@@ -32,7 +32,7 @@
 import pwd
 from eucadmin.command import Command
 
-RootWrapPath = 'usr/lib/eucalyptus/euca_rootwrap'
+RootWrapPath = '/usr/libexec/eucalyptus/euca_rootwrap'
 
 class Check(object):
 
diff -ur eucalyptus-3.1.0.orig/clc/eucadmin/eucadmin/eucasetup.py eucalyptus-3.1.0/clc/eucadmin/eucadmin/eucasetup.py
--- eucalyptus-3.1.0.orig/clc/eucadmin/eucadmin/eucasetup.py	2012-06-19 14:41:48.000000000 -0400
+++ eucalyptus-3.1.0/clc/eucadmin/eucadmin/eucasetup.py	2012-08-16 16:32:54.400888899 -0400
@@ -32,7 +32,7 @@
 import pwd
 from eucadmin.utils import chown_recursive, chmod_recursive
 
-RootWrapPath = 	'usr/lib/eucalyptus/euca_rootwrap'
+RootWrapPath = 	'/usr/libexec/eucalyptus/euca_rootwrap'
 MountWrapPath = 'usr/lib/eucalyptus/euca_mountwrap'
 
 MakeDirs = ['var/lib/eucalyptus/dynserv/data',
