--- eucalyptus-3.1.0.orig/clc/modules/postgresql/conf/scripts/setup_db.groovy	2012-08-18 10:13:43.277478917 -0400
+++ eucalyptus-3.1.0/clc/modules/postgresql/conf/scripts/setup_db.groovy	2012-08-22 19:15:31.930891442 -0400
@@ -79,7 +79,8 @@
 import groovy.sql.Sql
 import org.apache.log4j.Logger
 import org.logicalcobwebs.proxool.ProxoolFacade
-
+import java.io.FileWriter
+import java.lang.Appendable
 
 import static com.google.common.io.Closeables.closeQuietly
 import static com.google.common.io.Flushables.flushQuietly
@@ -130,14 +131,14 @@
     private static long   MIN_SHMMAX = 536870912L //512MB
 
     private int runProcess( List<String> args ) {
-        PrintStream outstream = null
-        PrintStream errstream = null
+        Appendable outstream = null
+        Appendable errstream = null
         LOG.debug("Postgres 9.1 command : " + args.toString( ) )
         try {
             ProcessBuilder pb = new ProcessBuilder(args)
             Process p = pb.start()
-            outstream = new PrintStream( BaseDirectory.LOG.getChildFile("db.log") )
-            errstream = new PrintStream( BaseDirectory.LOG.getChildFile("db-err.log") )
+            outstream = new FileWriter( BaseDirectory.LOG.getChildFile("db.log"), true )
+            errstream = new FileWriter( BaseDirectory.LOG.getChildFile("db-err.log"), true )
             p.consumeProcessOutput(outstream, errstream)
             int result = p.waitFor()
             flushQuietly( outstream )
@@ -333,7 +331,7 @@
         try {
             final Map<String,String> requiredProperties = [
                     max_connections: '8192',
-                    unix_socket_directory: "'" + SubDirectory.DB.getChildPath( EUCA_DB_DIR ) + "'",
+                    unix_socket_directories: "'" + SubDirectory.DB.getChildPath( EUCA_DB_DIR ) + "'",
                     ssl: PG_USE_SSL ? 'on' : 'off',
                     ssl_ciphers: '\'AES128-SHA:AES256-SHA\''
             ]
